kotlin/src/main/com/looker/sdk/ApiSettings.kt

```diff
@@ -49,76 +35,101 @@ open class ApiSettings(val rawReadConfig: () -> Map<String, String>) : Configura
             if (settings.isNullOrEmpty()) {
-                throw Error("No section named '$first_section' was found")
+                throw Error("No section named '$firstSection' was found")
             }
 
-            return ApiSettings({ settings })
+            return ApiSettings { settings }
         }
 
         fun fromMap(config: Map<String, String>): ConfigurationProvider {
-            return ApiSettings({ config })
+            return ApiSettings { config }
         }
     }
 
     override var baseUrl: String = ""
-    override var apiVersion: String = DEFAULT_API_VERSION
+    override var environmentPrefix: String = ENVIRONMENT_PREFIX
+    override var apiVersion: String = API_VERSION
+    override var headers: Map<String, String> = mapOf(LOOKER_APPID to AGENT_TAG, "User-Agent" to AGENT_TAG)
     override var verifySSL: Boolean = true
     override var timeout: Int = DEFAULT_TIMEOUT
-    override var headers: Map<String, String> = mapOf()
+
+    private val keyBaseUrl: String = "base_url"
+    private val keyApiVersion: String = "api_version"
+    private val keyEnvironmentPrefix: String = "environmentPrefix"
+    private val keyVerifySSL: String = "verify_ssl"
+    private val keyTimeout: String = "timeout"
 
     init {
-        val settings = rawReadConfig()
+        val settings = this.readConfig()
 
         // Only replace the current values if new values are provided
-        settings["base_url"].let { value ->
+        settings[keyBaseUrl].let { value ->
             baseUrl = unQuote(value ?: baseUrl)
         }
 
-        settings["api_version"].let { value ->
+        settings[keyApiVersion].let { value ->
             apiVersion = unQuote(value ?: apiVersion)
         }
 
-        settings["verify_ssl"].let { value ->
+        settings[keyEnvironmentPrefix].let { value ->
+            environmentPrefix = unQuote(value ?: environmentPrefix)
+        }
+
+        settings[keyVerifySSL].let { value ->
             verifySSL = asBoolean(value) ?: verifySSL
         }
-        settings["timeout"].let { value ->
+
+        settings[keyTimeout].let { value ->
             timeout = if (value !== null) value.toInt() else timeout
         }
+
     }
 
     override fun isConfigured(): Boolean {
         return baseUrl.isNotEmpty() && apiVersion.isNotEmpty()
     }
 
     override fun readConfig(): Map<String, String> {
-        // Merge any provided settings with the calculated values for the TransportOptions
-        return rawReadConfig().plus(
-            mapOf(
-                "base_url" to baseUrl,
-                "api_version" to apiVersion,
-                "verify_ssl" to verifySSL.toString(),
-                "timeout" to timeout.toString(),
+        // Load environment variables and possibly overwrite with explicitly declared map values
+        val rawMap = readEnvironment().plus(rawReadConfig())
+        // TODO environment variables must overwrite or append to rawMap values
```

Comment:
Is this TODO still relevant?  Looks like there's some environment code below.

Reply:
It's not. Removing.